pkg/apis/machine/validation/machinedeployment_test.go

```diff
@@ -0,0 +1,183 @@
+			}))))
+		})
+
+		It("should return error if machine template spec class kind is not defined", func() {
+			machineDeployment.Spec.Template.Spec.Class.Kind = ""
+
+			Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":   Equal(field.ErrorTypeRequired),
+				"Field":  Equal("spec.template.spec.class.kind"),
+				"Detail": Equal("Kind is required"),
+			}))))
+		})
+
+		It("should return error if machine template spec class name is not defined", func() {
+			machineDeployment.Spec.Template.Spec.Class.Name = ""
+
+			Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":   Equal(field.ErrorTypeRequired),
+				"Field":  Equal("spec.template.spec.class.name"),
+				"Detail": Equal("Name is required"),
+			}))))
+		})
+
+		Context("Validate UpdateStrategy of machine deployment", func() {
+			It("should return error if strategy type is not valid", func() {
+				machineDeployment.Spec.Strategy.Type = "invalid"
+
+				Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(field.ErrorTypeInvalid),
+					"Field":  Equal("spec.strategy.type"),
+					"Detail": Equal("strategy type must be one of [InPlaceUpdate Recreate RollingUpdate]"),
+				}))))
+			})
+
+			Context("Validate RollingUpdate strategy", func() {
+				It("should not return error if RollingUpdate strategy is valid", func() {
+					Expect(ValidateMachineDeployment(machineDeployment)).To(BeEmpty())
+				})
+
+				It("should return error if MaxUnavailable is not valid", func() {
+					machineDeployment.Spec.Strategy.RollingUpdate.UpdateConfiguration.MaxUnavailable = nil
+
+					Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":   Equal(field.ErrorTypeRequired),
+						"Field":  Equal("spec.strategy.rollingUpdate.maxUnavailable"),
+						"Detail": Equal("unable to convert maxUnavailable to int32"),
+					}))))
+				})
+
+				It("should return error if MaxSurge is not valid", func() {
+					machineDeployment.Spec.Strategy.RollingUpdate.UpdateConfiguration.MaxSurge = nil
+
+					Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":   Equal(field.ErrorTypeRequired),
+						"Field":  Equal("spec.strategy.rollingUpdate.maxSurge"),
+						"Detail": Equal("unable to convert maxSurge to int32"),
+					}))))
+				})
+			})
+
+			Context("Validate InPlaceUpdate strategy", func() {
+				BeforeEach(func() {
+					machineDeployment.Spec.Strategy = machine.MachineDeploymentStrategy{
+						Type: machine.InPlaceUpdateMachineDeploymentStrategyType,
+						InPlaceUpdate: &machine.InPlaceUpdateMachineDeployment{
+							UpdateConfiguration: machine.UpdateConfiguration{
+								MaxUnavailable: ptr.To(intstr.FromInt(1)),
```

Comment:
Nit: 
Since `intstr.FromInt()` has been deprecated, can you change this to use `FromInt32` instead? (ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/intstr#FromInt)

Reply:
Addressed in d63202a9a0d338cc702165bd619538a5ca4d27fd. PTAL